### PR TITLE
asserts: introduce snap-declaration assertion (SnapDeclaration), enough support to decode it

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
waiting on prereq for now

File: asserts/snap_asserts.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	_, err = checkMandatory(assert.headers, "snap-digest")
+	if err != nil {
+		return nil, err
+	}
+
+	grade, err := checkMandatory(assert.headers, "grade")
+	if err != nil {
+		return nil, err
+	}
+	if grade != "stable" && grade != "devel" {
```
Comment (niemeyer) [inline asserts/snap_asserts.go]: We probably don't need this here. This is a safety net for the store to not publish things the publisher isnt really intending to, so likely snappy won't even be looking at the data, which means there's probably no reason for us to break clients if we extend this somehow.

**Comment by niemeyer** [comment]
LGTM

**Comment by zyga** [comment]
Looks good to me. +1

**Pull request merged**